### PR TITLE
Send size to /avatars if requested

### LINKED ISSUE #15453: Gravatar resolution mismatch
- Gitea version (or commit ref): 1.14.0-rc2, 1.14.0
- Git version: 2.20.1
- Operating system: Linux 4.19.0-14-amd64 #1 SMP Debian 4.19.171-2 (2021-01-30) x86_64 GNU/Linux
- Binaries from: https://github.com/go-gitea/gitea/releases/download/
- Gitea is running with a systemd service file pretty similar to what is provided in the documentation, but on the default port with a nginx proxy infront.
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes https://try.gitea.io/test0r
  - [ ] No
- Log gist:

## Description
The avatars sourced by gravatar have a too small resolution which results in a pixelated gravatar avatar. This behaviour is also reporducible on https://try.gitea.io.
After some investigation it seems as gitea queries the default gravatar size of 80x80 pixel. Which means that the 80x80 pixel avatar is expanded to 260x260px. A diff between the default gravatar image with size 80x80 pixel and the avatar delivered by gitea does not gather any difference between those two files.

## Screenshots
2021-04-13_17:44:23_3000x1920

### PR DESCRIPTION
**Pull request by zeripath**
If an avatar is requested in a particular size ensure that /avatars also gets the
size request

Fix #15453

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: models/user_avatar.go [hunk 1]
```diff
@@ -82,14 +82,19 @@ func (u *User) RealSizedAvatarLink(size int) string {
 		if u.Avatar == "" {
 			return DefaultAvatarLink()
 		}
+		if size > 0 {
+			return setting.AppSubURL + "/avatars/" + u.Avatar + "?size=" + strconv.Itoa(size)
```
Comment (KN4CK3R) [inline models/user_avatar.go:86]: What does the `size` parameter do?

Comment (zeripath) [reply models/user_avatar.go:86]: nothing right now - but we should probably keep sending it for future options

**Comment by zeripath** [comment]
This PR means that if you request a size that size will get passed down to gravatar.

**Review by silverwind — APPROVED** [review]
Can confirm this fixes the issue with such a configuration:

```
[picture]
DISABLE_GRAVATAR = false
GRAVATAR_SOURCE = gravatar
ENABLE_FEDERATED_AVATAR = true
```

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Pull request merged**